client/app/models/package.js

```diff
@@ -90,30 +99,58 @@ export default class PackageModel extends Model {
   }
 
   async save(recordsToDelete) {
-    await this.fileManager.save();
-    if (this.dcpPackagetype === DCPPACKAGETYPE.PAS_PACKAGE.code) {
-      await this.saveDeletedRecords(recordsToDelete);
-      await this.pasForm.save();
-    }
-    if (this.dcpPackagetype === DCPPACKAGETYPE.RWCDS.code) {
-      await this.rwcdsForm.save();
-    }
-    if (this.dcpPackagetype === DCPPACKAGETYPE.DRAFT_LU_PACKAGE.code
-      || this.dcpPackagetype === DCPPACKAGETYPE.FILED_LU_PACKAGE.code) {
-      await this.saveDeletedRecords(recordsToDelete);
-      await this.landuseForm.save();
+    let formAdapterError = false;
+
+    try {
+      if (this.dcpPackagetype === DCPPACKAGETYPE.PAS_PACKAGE.code) {
+        await this.saveDeletedRecords(recordsToDelete);
+        await this.pasForm.save();
+      }
+      if (this.dcpPackagetype === DCPPACKAGETYPE.RWCDS.code) {
+        await this.rwcdsForm.save();
+      }
+      if (this.dcpPackagetype === DCPPACKAGETYPE.DRAFT_LU_PACKAGE.code
+        || this.dcpPackagetype === DCPPACKAGETYPE.FILED_LU_PACKAGE.code) {
+        await this.saveDeletedRecords(recordsToDelete);
+        await this.landuseForm.save();
+      }
+      if (this.dcpPackagetype === DCPPACKAGETYPE.FILED_EAS.code) {
+        await this.saveDirtySingleCeqrInvoiceQuestionnaire();
+      }
+      if (this.dcpPackagetype === DCPPACKAGETYPE.DRAFT_SCOPE_OF_WORK.code) {
+        await this.saveDirtySingleCeqrInvoiceQuestionnaire();
+      }
+    } catch (e) {
+      console.log('Error saving a Form or Ceqr Invoice Questionnaire: ', e);
+
+      formAdapterError = true;
     }
-    if (this.dcpPackagetype === DCPPACKAGETYPE.FILED_EAS.code) {
-      await this.saveDirtySingleCeqrInvoiceQuestionnaire();
+
+    try {
+      await super.save();
+    } catch (e) {
+      console.log('Error saving package: ', e);
     }
-    if (this.dcpPackagetype === DCPPACKAGETYPE.DRAFT_SCOPE_OF_WORK.code) {
-      await this.saveDirtySingleCeqrInvoiceQuestionnaire();
+
+    try {
+      await this.fileManager.save();
+    } catch (e) {
```

Comment:
Why do we need 3 separate try-catches?

Reply:
I wanted to improve the user experience of Save. Previously if a form save() failed, the package and files would not save, and vice versa. I wanted to make it so that if a form save failed, files would still upload.  The try/catch around each save() makes sure that execution continues even if a failure occurs for a previous save().

Reply:
Are you saying that if a form save fails (e.g. a user entered something into a text input and that value was NOT PATCHed to the database) that you still want files to be uploaded? Can you explain more about why we want this behavior?

If another part of the form fails to save, wouldn't we also want the document upload to fail simultaneously? Everything functioning in sync

Reply:
Yeah, that is a very good counterpoint. Here are the pros and cons I can think of for allowing execution to continue after an error with one of the three saves.

(Note that with this PR, the current order of operations is.. 
     1) form saves 2) package saves 3) files save
Previously, it was 
     1) files save 2) form saves 3) package save

I re-ordered the operations to match the order in the UI)

Pros: 
   1) the user would find it less bothersome to have to re-select many files if an error occurs on form or package save. 
   2) without the try/catch, it's possible that  a partial save occurs anyway. E.g. the first save operation for forms could have succeeded, but then the second one for files fails. With the try/catch, it just means that order doesn't affect what gets saved and what doesn't.

Cons:
  1) This introduces a lot of extra error handling code, as evidenced in this PR. We have to manually prevent some operations from executing if any errors occurred.